src/bedrock/common/Stream.js

```diff
@@ -0,0 +1,364 @@
+const MAX_ALLOC_SIZE = 1024 * 1024 * 2 // 2MB
+const DEFAULT_ALLOC_SIZE = 10000
+
+class ByteStream {
```

Comment:
this looks a lot like protodef..

Reply:
Yeah, but I needed zigzag varints, peek and buffer concatenation functions. The original lib used in bedrock-provider was the JSPrismarine/BinaryStream class, but that did a bunch of unnecessary slow assertions on top of the ones in the node Buffer class